bundles/SeoBundle/public/js/startup.js

```diff
@@ -0,0 +1,96 @@
+pimcore.registerNS("pimcore.seo");
+
+
+pimcore.seo = Class.create({
+    initialize: function () {
+        // TODO mattamon add key event once the new event is merged
+        //document.addEventListener(pimcore.events.preRegisterKeyBindings, this.registerKeyBinding.bind(this));
+        document.addEventListener(pimcore.events.pimcoreReady, this.pimcoreReady.bind(this));
+    },
+
+    pimcoreReady: function (e) {
+        const user = pimcore.globalmanager.get('user');
+        const perspectiveCfg = pimcore.globalmanager.get("perspective");
+
+        if (perspectiveCfg.inToolbar("marketing") && perspectiveCfg.inToolbar("marketing.seo")) {
+            var seoMenu = [];
+
+            if (user.isAllowed("documents") && user.isAllowed("seo_document_editor") && perspectiveCfg.inToolbar("marketing.seo.documents")) {
+                seoMenu.push({
+                    text: t("seo_document_editor"),
+                    iconCls: "pimcore_nav_icon_document_seo",
+                    itemId: 'pimcore_menu_marketing_seo_document_editor',
+                    handler: this.showDocumentSeo
+                });
+            }
+
+            if (user.isAllowed("robots.txt") && perspectiveCfg.inToolbar("marketing.seo.robots")) {
+                seoMenu.push({
+                    text: t("robots.txt"),
+                    iconCls: "pimcore_nav_icon_robots",
+                    itemId: 'pimcore_menu_marketing_seo_robots_txt',
+                    handler: this.showRobotsTxt
+                });
+            }
+
+            if (user.isAllowed("http_errors") && perspectiveCfg.inToolbar("marketing.seo.httperrors")) {
+                seoMenu.push({
+                    text: t("http_errors"),
+                    iconCls: "pimcore_nav_icon_httperrorlog",
+                    itemId: 'pimcore_menu_marketing_seo_http_errors',
+                    handler: this.showHttpErrorLog
+                });
+            }
+
+            // get index of marketing.targeting
+            if (seoMenu.length > 0) {
+                const toolbar = pimcore.globalmanager.get('layout_toolbar');
+
+                toolbar.marketingMenu.add({
```

Comment:
You are right, maybe we should discuss, what should happen here then. 
There are some changes with the PimcoreGlossaryBundle but they still would not solve this problem.
https://github.com/pimcore/pimcore/pull/13811/files#diff-0bd635bc68ae1ed97161f850140cef5707f9be71d5dd982c1f978c6aab2255c7R604

Maybe we should add those menus even though they are empty and check before creating the menu again.
https://github.com/pimcore/pimcore/pull/13811/files#diff-0bd635bc68ae1ed97161f850140cef5707f9be71d5dd982c1f978c6aab2255c7R1156 -> with this event you can still add items. Maybe we have a check after that if some menu items would be empty right? 

Wdyt?

Reply:
Like we discussed in the daily:
https://github.com/pimcore/pimcore/pull/13811/files#diff-0bd635bc68ae1ed97161f850140cef5707f9be71d5dd982c1f978c6aab2255c7R482 -> i removed the if condition.
Everyone who does the refactoring should do this, so we do it one by one.

Unfortunately it is not possible to simply hide the menu item so I do not create it if items are empty.
https://github.com/pimcore/pimcore/pull/13811/files#diff-0bd635bc68ae1ed97161f850140cef5707f9be71d5dd982c1f978c6aab2255c7R1164

Reply:
what about moving all the code in a separate method 
eg. 
```
createMarketingMenu: function(items){
   this.marketingMenu = new Ext.menu.Menu({
   ...
}
```
and from the bundle, check if `typeof toolbar.marketingMenu == 'undefined` , then toolbar.createMarketingMenu, else  `toolbar.marketingMenu.add();`

Reply:
Like I mentioned, I refactored everything in the GlossaryBundle, so maybe we should wait for those changes to be merged. :) This add method won't work anymore because we create every menu at the very end, to avoid sorting the menu after it is already rendered.